protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/AuthCmdHandler.java

```diff
@@ -339,6 +342,31 @@ private Response doLoginAuthPassCheck(SMTPSession session, Username username, St
         return doAuthTest(session, username, pass, "LOGIN");
     }
 
+    protected Response doDelegation(SMTPSession session, Username username) {
+        Response res = null;
+
+        List<AuthHook> hooks = Optional.ofNullable(getHooks())
+            .orElse(List.of());
+
+        for (AuthHook rawHook : hooks) {
+            rawHook.doDelegation(session, username);
+            res = executeHook(session, rawHook, hook -> rawHook.doDelegation(session, username));
+
+            if (res != null) {
+                if (SMTPRetCode.AUTH_FAILED.equals(res.getRetCode())) {
+                    LOGGER.warn("DELEGATE failed from {}", username);
+                } else if (SMTPRetCode.AUTH_OK.equals(res.getRetCode())) {
+                    LOGGER.debug("DELEGATE succeed from {}", username);
```

Comment:
It would be nice to know who we did delegate to...

Reply:
info level for the success...

```suggestion
                    LOGGER.warn("{} was not authorized to connect as {}", session.getUser(), username);
                } else if (SMTPRetCode.AUTH_OK.equals(res.getRetCode())) {
                    LOGGER.info("{} was authorized to connect as {}", session.getUser(), username);
```